tools/lib/h5trav.c

```diff
@@ -281,13 +281,13 @@ traverse(hid_t file_id, const char *grp_name, hbool_t visit_start, hbool_t recur
             /* Visit all links in group, recursively */
             if (H5Lvisit_by_name2(file_id, grp_name, trav_index_by, trav_index_order, traverse_cb, &udata,
                                   H5P_DEFAULT) < 0)
-                H5TOOLS_GOTO_ERROR((-1), "H5Lvisit_by_name failed");
+                H5TOOLS_ERROR((-1), "H5Lvisit_by_name failed");
         } /* end if */
         else {
             /* Iterate over links in group */
             if (H5Literate_by_name2(file_id, grp_name, trav_index_by, trav_index_order, NULL, traverse_cb,
                                     &udata, H5P_DEFAULT) < 0)
-                H5TOOLS_GOTO_ERROR((-1), "H5Literate_by_name failed");
+                H5TOOLS_ERROR((-1), "H5Literate_by_name failed");
         } /* end else */
```

Comment:
This will continue the traversal, even if there was an iteration problem, which I think is better than immediately jumping to the error condition. In the latter case, we could have moved the cleanup to after the done target.